### ISSUE TITLE
508-defect-2:[SCREENREADER]: Image of map and image of vet center MUST have descriptive and meaningful alternative text OR not be announced if repetitive

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#25011]
## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user or a user with low bandwidth, I want to be able to read descriptive and meaningful alternative text to images so I can have an equivalent experience.

## Accessibility feedback
<details>
<summary> Collapsed details </summary>

## Details

The image of the vet center has unhelpful alt text "The entrance to the Escanaba Vet Center" that provides nothing of value. It also contains a repetitive link to the same google map in the earlier link.

The image of the map has generic alt text that announces, "Image, static map" despite providing important contextual location like street names. It also links to directions to how to get to the vet center on google maps which makes for three separate links that do the same thing.

## Acceptance Criteria and Recommendations

Depending on the purpose of the image of the Vet Center for sighted users or users with high bandwidth, the alt text should..

- [ ] Have an empty alt text e.g. `alt=""` if it is meant only to provide superficial aesthetic value
- [ ] Describe the image of the Vet Center purposefully e.g. "Escanaba Vet Center is a large brick building located between the Shoppers and Petco."

Depending on the purpose of the static map for sighted users high bandwidth, the alt text should...

- [ ] Have an empty alt text e.g. `alt=""` if it is only meant to link to directions on google maps OR
- [ ] Describe the image of the map e.g. "Escanaba Vet Center is located on Ludington Street" if it is meant to provide key contextual clues to its location via a visual map

The designer should:
- [ ] determine the intent of the image and static map and choose between the acceptance criteria with guidance from an accessibility specialist
- [ ] consider if image, static map, and "get directions" should all share the same link or have links to begin with

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/14154792/119001784-48880100-b95a-11eb-9ce2-d5dbf437ca25.mov

Both alt text descriptions have no value to screen reader users and do not provide an equivalent experience.

<img width="729" alt="Screen Shot 2021-05-20 at 10 59 13 AM" src="https://user-images.githubusercontent.com/14154792/119002100-8f75f680-b95a-11eb-956b-52cd64575ff6.png">

<img width="1468" alt="Screen Shot 2021-05-20 at 11 10 02 AM" src="https://user-images.githubusercontent.com/14154792/119003844-0f509080-b95c-11eb-99ae-9467cc0d36af.png">

</details>

## Decisions
- Since the facility image is intended to assist in wayfinding, we will add guidance to the content deck to support editors in using alt text to include concise meaningful details to help Veterans find the facility for the first time - supporting all users (regardless of ability/assistive tech or bandwidth). 
- We will reinforce the use of the "Prepare for your visit" accordions to offer other important information about parking, signage, or landmarks as appropriate.
- Map: The map can be hidden from screenreaders since an equivalent experience is offered via the "Get Directions" link

## Tasks
- [ ] Hide map from screenreaders 

## Dev notes
The map can be hidden by using `aria-hidden="true"`

## Acceptance Criteria
- [ ] The map slice is not announced to Veterans using screen readers.

**Comment by lildziner** [comment]
Hi @joshkimux @mmiddaugh I have some follow-up thoughts and questions:

## Facility Image

### Link to Google maps:

* I agree that link should be removed
* The facilities team discussed a future iteration where the image would click to an enlarged version but we would want to plan for that soon so that content editors can upload a photo with the enlarged size as the minimum requirement.  

### Alt text:

Veterans found this image to be very valuable and they drew really unique conclusions that would seem to contribute to their decision making process for choosing a facility. I don’t know if a non-sighted would find this equally valuable but my assumption is yes. Quoting from memory some of the things we heard in research:

* “I can see there’s parking in front of the building”,
* “It looks like its in a strip mall which means it’s pretty convenient”,
* “It looks like its in a strip mall so I’d probably not want to go there for therapy”
* ”I like that there are plants in front of the entrance. It makes it look friendly”

@mmiddaugh, do we think this type of direction could be added to the CMS UI via hint text/example copy to explain why and how to provide some of this intangible information? 



## Static map image

### Link to Google maps:

- In user tests, users clicked on both the map image and 'Get directions' link. I don’t recall negative feedback or that the experience was unexpected. So, I don’t think the redundancy is a problem for a sighted user. @joshkimux Is it still problematic if **two** links go to Google maps?

### Alt text:

- I think this could be empty as it would likely be duplicative of the info found in “Directions and Parking” accordion, eg. “The Orlando Vet Center is located in the back, inside corner of the TerraCotta Business Park behind the Taco Bell and Checkers restaurants. Use any Vet Center designated parking spot.”

**Comment by mmiddaugh** [comment]
thanks @joshkimux and @AngelaFowler82 - 

My alt text description for the image of Escanaba Vet Center was based on guidance previously provided in #4255: 

> "The recommendation is to have the alt text read "Photo of Pittsburgh VAMC University Drive Facility", so that the screen reader reads as "Photo of Pittsburgh VAMC University Drive Facility, image". A screen reader user may be able to say to someone else, who may accompany them, that there is a photo of the facility. In this context, the detail that it is a photo adds useful context."

As @lildziner suggests, we can provide content recommendations/examples to suggest editors add details about the pictured facility - in the content deck, at a minimum. Currently, the CMS UI requires alt text and indicates editors should add a "short description of the image to be used by screen readers and displayed when the image isn't loaded". 

Our content guidance recommends alt text be 125 characters or less to be read by screen readers. Is this correct, @AngelaFowler82? 

The alt text should not be a blocker as it will be content controlled by the field, although we can give examples, recommendations, and best practices to encourage editors to provide thoughtful and meaningful descriptions.

**Comment by mmiddaugh** [comment]
This issue will need to be discussed with OCTO-DE to determine the appropriate user experience across the products
@davidconlon

**Comment by mmiddaugh** [comment]
@davidconlon suggestions/clarification from an earlier conversation with Angela today: 

Image: Since the facility image is intended to assist in wayfinding, we can add guidance to the content deck to support editors in using alt text to include concise meaningful details to help Veterans find the facility for the first time - supporting all users (regardless of ability/assistive tech or bandwidth). We can also reinforce the use of the "Prepare for your visit" accordions to offer other important information about parking, signage, or landmarks as appropriate. 

 Map: The map can be hidden from screenreaders since an equivalent experience is offered via the "Get Directions" link.

**Comment by mmiddaugh** [comment]
Will have @AngelaFowler82 validate this issue once we have a Vet Center published and available on Staging.

**Comment by mmiddaugh** [comment]
This can now be validated in Prod 🎉 
https://www.va.gov/nashville-vet-center/
https://www.va.gov/colorado-springs-vet-center/
@AngelaFowler82 @joshkimux

**Comment by AngelaFowler82** [comment]
You guys have made dramatic improvements to not only how the maps and images are handled but the experience of determining how to get to the vet center as a blind person. My only suggestion would be for the image of the vet center to render under the main location heading rather than under the list of other locations.

**Comment by joshkimux** [comment]
@mmiddaugh could we open up a new ticket to address Angela's comment on headings? Right now the image of the vet center is in other locations, which gives the illusion that it's related to other locations instead of the main location itself.

**Comment by mmiddaugh** [comment]
I apologize! I meant to ask a few clarifying questions before opening an issue. 
I appreciate if you are able to clarify directly in the description of the new issue: #27497
